src/Text/Pretty/Simple/Internal/Printer.hs

```diff
@@ -255,7 +255,8 @@ prettyExpr opts = (if outputOptionsCompact opts then group else id) . \case
             spaceIfNeeded = \case
               Other (' ' : _) : _ -> mempty
               _ -> space
-    lineAndCommaSep x y = x <> line' <> annotate Comma "," <> y
+    lineAndCommaSep x y = x <> munless (outputOptionsCompact opts) line' <> annotate Comma "," <> y
+    munless b x = if b then mempty else x
```

Comment:
I've been meaning for a while to propose that `mwhen`/`munless` be added to `base` or `extra`. I end up using them all the time...

Reply:
Nice